service/orchestrator/cloud_service.go

```diff
@@ -58,31 +60,39 @@ func (s *Service) RegisterCloudService(_ context.Context, req *orchestrator.Regi
 	service.Description = req.Service.Description
 
 	// Persist the service in our database
-	s.db.Create(&service)
+	err = s.storage.Create(&service)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not add cloud service to the database: %v", err)
+	}
 
 	return
 }
 
+// ListCloudServices implements method for OrchestratorServer interface for listing all cloud services
 func (s *Service) ListCloudServices(_ context.Context, _ *orchestrator.ListCloudServicesRequest) (response *orchestrator.ListCloudServicesResponse, err error) {
 	response = new(orchestrator.ListCloudServicesResponse)
 	response.Services = make([]*orchestrator.CloudService, 0)
 
-	err = s.db.Find(&response.Services).Error
+	err = s.storage.List(&response.Services)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "database error: %s", err)
 	}
 
 	return response, nil
 }
 
+// GetCloudService implements method for OrchestratorServer interface for getting a cloud service with provided id
 func (s *Service) GetCloudService(_ context.Context, req *orchestrator.GetCloudServiceRequest) (response *orchestrator.CloudService, err error) {
-	if req == nil || req.ServiceId == "" {
-		return nil, status.Errorf(codes.InvalidArgument, "service id is empty")
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, orchestrator.ErrRequestIsNil.Error())
+	}
+	if req.ServiceId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, orchestrator.ErrIDIsMissing.Error())
 	}
 
 	response = new(orchestrator.CloudService)
-	err = s.db.First(&response, "Id = ?", req.ServiceId).Error
-	if errors.Is(err, gorm.ErrRecordNotFound) {
+	err = s.storage.Get(&response, "Id = ?", req.ServiceId)
```

Comment:
This creates an unnecessary pointer pointer, since service is already `*CloudService`